ErrorReporting/src/Bootstrap.php

```diff
@@ -130,6 +130,14 @@ public static function exceptionHandler($ex)
             $service = self::$psrLogger->getMetadataProvider()->serviceId();
             $version = self::$psrLogger->getMetadataProvider()->versionId();
             self::$psrLogger->error($message, [
+                'context' => [
```

Comment:
Are there published docs for what these options are? Should these options be added to the `PsrLogger`'s documentation?

Reply:
So, looking at this a bit more I think there may be some additional changes we need. I'm not completely familiar with the interactions between Stackdriver Logging/Error Reporting, but the PSR logger should work as follows:

```php
$psrLogger->error('Testing: {message}', [
    'message' => 'my message', // should fill the message placeholder above (e.g. "Testing: my message")
    'anyOtherKey' => 'test' // this will be appended to the LogEntry's jsonPayload (https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry)
    'stackdriverOptions' => [ // this key is reserved for stack driver specific options, which I believe is what we want to be using for reporting the filePath/lineNumber
        'labels' => [
            'my' => 'label'
        ]
    ]
]);
```

With this current code, I believe `context` will be appended to the `jsonPayload` of a log entry, is that the desired effect?

After testing with this:

```php
require 'vendor/autoload.php';

use Google\Cloud\ErrorReporting\Bootstrap;

Bootstrap::init();

throw new \Exception('testing');
```

I only get a log showing up in my Stackdriver Logging viewer in the cloud console. Should it be accessible through the Error Reporting console? If so, it seems like we want to trigger [this RPC](https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events/report#ReportedErrorEvent) as opposed [to writing a log entry](https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/write)? Or is there some process that imports log entries from stackdriver into error reporting?

Reply:
@dwsupplee So there is  strange behavior in the StackDriver Logging API that allows logs to be automatically detected as Errors and so sent to the Error Reporting UI. The reason your example above didn't work is because the proper environment variables are not set. The automatic error reporting behavior I am implementing is for App Engine specifically, and unless the [MetadataProvider](https://github.com/GoogleCloudPlatform/google-cloud-php/blob/master/Core/src/Report/MetadataProviderUtils.php) is set and populates the `service` and `version` properties, the logged error is not sent into Error Reporting.

A larger refactoring of this may be necessary. Maybe we should send exceptions / errors to the Error Reporting API and skip sending them to the Logging API. The reason we DIDN'T do this is because gRPC was required for Error Reporting, but as that is no longer the case I am definitely on board with switching.

Personally, I'm in favor of merging this small fix now (in anticipation of App Engine for PHP 7.2) and I can submit a more inclusive fix later. But if you're not in favor of the quick fix I can close this PR.

Reply:
That makes perfect sense, thanks for the explanation :). The fix LGTM given the context!

Reply:
It looks like this matches up with the [ErrorContext](https://cloud.google.com/error-reporting/reference/rest/v1beta1/ErrorContext) object in the Error Reporting API.